### PR TITLE
Implement intersection check for polygon

### LINKED ISSUE #37: Implement Polygon::intersects(other: &Polygon)
Use [SAT](http://www.dyn4j.org/2010/01/sat/). Requires the following functions:

- [x] Polygon can convert its edges to vectors
- [x] Vectors can return their normals

### PR DESCRIPTION
**Pull request by janhohenheim**
Resolves #37

### CODE REVIEW

File: geometry/src/polygon.rs [hunk 1]
```diff
@@ -547,7 +602,144 @@ mod tests {
...
+            Point { x: 0.0, y: -20.0 },
+            Point { x: 0.0, y: 2.0 },
+        ])
+        .unwrap();
+        assert!(first_polygon.intersects(&second_polygon));
+        assert!(second_polygon.intersects(&first_polygon));
+    }
+
+    #[test]
+    fn does_not_intersect_when_appart() {
```
Comment (jeremystucki) [inline geometry/src/polygon.rs]: *apart

**Review by jeremystucki — CHANGES_REQUESTED** [review]

File: geometry/src/polygon.rs [hunk 2]
```diff
@@ -138,15 +139,69 @@ impl Polygon {
...
+    fn scalar_project_onto_unit_vector(&self, axis: Vector) -> (f64, f64) {
+        let projection: Vec<_> = self
+            .vertices()
+            .iter()
+            .cloned()
+            .map(Vector::from)
+            .map(|vector| vector.dot_product(axis))
+            .collect();
+        (
+            *projection
```
Comment (jeremystucki) [inline geometry/src/polygon.rs:161]: Could you sort this and then take the first and last elements?

**Review by jeremystucki — CHANGES_REQUESTED** [review]

File: geometry/src/polygon.rs (hunk 2 above)
Comment (janhohenheim) [reply geometry/src/polygon.rs:161]: calling min(n) and max(n) is a O(n) algorithm. Quicksort is O(n log n). O(n) is more efficient: https://www.wolframalpha.com/input/?i=f(x)+%3D+x+and+g(x)+%3D+x+*+log+x+for+x+%3D+0+to+x+%3D+100

File: geometry/src/polygon.rs (hunk 1 above)
Comment (janhohenheim) [reply geometry/src/polygon.rs]: Addressed in https://github.com/myelin-ai/engine/pull/53/commits/76d991b7b385fb91ab9ede5effc393f0e65b9003

**Review by jeremystucki — APPROVED** [review]

**Pull request merged**